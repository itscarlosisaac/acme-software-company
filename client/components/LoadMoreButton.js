import React, { Component } from 'react'
import PropTypes from 'prop-types';
export default class LoadMoreButton extends Component {
  render() {
    return (
      <div className="app__load__more">
        <button onClick={this.props.handleLoadMore}>
          Load More Gifs
        </button>
      </div>
    )
  }
}

LoadMoreButton.propTypes = {
  handleLoadMore: PropTypes.func.isRequired,
}