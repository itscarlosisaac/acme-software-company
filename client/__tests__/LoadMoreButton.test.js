import React from 'react';
import LoadMoreButton from '../components/LoadMoreButton';

import { configure, shallow } from 'enzyme';
import Adapter from  'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({adapter: new Adapter() });

describe('<LoadMoreButton>', () => {
  it('Should render component', () => {
    const wrapper = shallow(<LoadMoreButton handleLoadMore={()=> {}} />)
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('button').text()).toBe('Load More Gifs');
    expect(wrapper.find('div').hasClass('app__load__more')).toBe(true);
  });

  it(`Should have a function as a prop`, () => {
    const wrapper = shallow(<LoadMoreButton handleLoadMore={()=> {}} />)
    const f = typeof wrapper.instance().props.handleLoadMore;
    expect(f).toBe('function')
  })
  it(`Matches the snapshot`, () => {
    const tree = shallow(<LoadMoreButton handleLoadMore={()=>{}} />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});