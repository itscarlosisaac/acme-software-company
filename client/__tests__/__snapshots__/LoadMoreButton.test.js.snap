// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadMoreButton> Matches the snapshot 1`] = `
<div
  className="app__load__more"
>
  <button
    onClick={[Function]}
  >
    Load More Gifs
  </button>
</div>
`;
